### ISSUE TITLE
rdkafka input plugin

### ISSUE THREAD

**Issue by repeatedly** [fluent/fluent-plugin-kafka#350]
rdkafka-ruby is now considering `each_batch` support. 

https://github.com/appsignal/rdkafka-ruby/pull/124

We can implement `in_rdkafka` with this feature.

**Comment by kenton18** [comment]
Hi author, may I ask that when can be release, I am really need this right now!!!

**Comment by kenton18** [comment]
I would like to try the beta version if it is possible

**Comment by repeatedly** [comment]
> may I ask that when can be release?

It depends on rdkafka-ruby. When rdkafka-ruby merge above patch and release new version, we can test it.

**Comment by tkornai** [comment]
Hi @repeatedly,

may I ask what causes the hard dependency on the above PR? I think a similar batching method could be fairly easily implemented in this client lib with the existing functionality exposed by rdkafka-ruby, which could be replaced in the future once the feature is merged.

**Comment by repeatedly** [comment]
I'm not familiar with entire rdkafka's API, so I can't judge above unfinished implementation is good or not.
If you ok with current implemenation, could you send a patch?

**Comment by tkornai** [comment]
I smoke tested the above PR locally, but it is definitely not battle hardened yet.

**Comment by tkornai** [comment]
I have created a new PR which helps with tracking the changes against the in_kafka_group.rb implementation:
https://github.com/fluent/fluent-plugin-kafka/pull/367

**Comment by kenhys** [comment]
released as fluent-plugin-kafka v0.15.0
